engine/random/c_src/rng.c

```diff
@@ -28,13 +27,21 @@ uint8_t os_random_secrandom(uint8_t *buf, size_t len)
 #elif defined(USE_SECRANDOM)
     return SecRandomCopyBytes(kSecRandomDefault, len, buf) == errSecSuccess ? 0 : 1;
 #elif defined(USE_CRYPTGENRANDOM)
-    HCRYPTPROV rng;
-    if (CryptAcquireContext(&rng, NULL, NULL, PROV_RSA_FULL, CRYPT_VERIFYCONTEXT | CRYPT_SILENT) == 0)
+    BCRYPT_ALG_HANDLE algo_handle;
+    NTSTATUS ret1 = BCryptOpenAlgorithmProvider(&algo_handle, BCRYPT_RNG_ALGORITHM,
```

Comment:
Is it necessary to initialize the provider if `algo_handle` is not being passed to `BCryptGenRandom`?

Reply:
Did I forget to pass it?

Reply:
> Did I forget to pass it?

Nope, I think it's just not needed here. The [docs](https://docs.microsoft.com/en-us/windows/win32/api/bcrypt/nf-bcrypt-bcryptgenrandom) say the `hAlgorithm` parameter should be `NULL` when using `BCRYPT_USE_SYSTEM_PREFERRED_RNG` so it seems like you don't need a handle at all

Reply:
Figured that one out after I wrote it.  Had it removed in an earlier version but added it back in by accident but forgot to actually put it in as a argument.